test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperTextBoxTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using System.ComponentModel.DataAnnotations;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    public class HtmlHelperTextBoxTest
+    {
+        private const string InputWithPlaceholderAttribute = @"<input id=""HtmlEncode[[Property1]]"" name=""HtmlEncode[[Property1]]"" placeholder=""HtmlEncode[[placeholder]]"" type=""HtmlEncode[[{0}]]"" value="""" />";
+        private const string InputWithoutPlaceholderAttribute = @"<input id=""HtmlEncode[[Property1]]"" name=""HtmlEncode[[Property1]]"" type=""HtmlEncode[[{0}]]"" value="""" />";
+
+        [Theory]
+        [InlineData("text")]
+        [InlineData("search")]
+        [InlineData("url")]
+        [InlineData("tel")]
+        [InlineData("email")]
+        [InlineData("number")]
+        public void TextBoxFor_GeneratesPlaceholderAttribute_WhenDisplayAttributePromptIsSetAndTypeIsValid(string type)
+        {
+            // Arrange            
+            var model = new TextBoxModel();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model);
+
+            // Act
+            var result = helper.TextBoxFor(m => m.Property1, new { type });
+
+            // Assert
+            Assert.Equal(string.Format(InputWithPlaceholderAttribute, type), HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        [Theory]
+        [InlineData("hidden")]
+        [InlineData("date")]
+        [InlineData("time")]
+        [InlineData("range")]
+        [InlineData("color")]
+        [InlineData("checkbox")]
+        [InlineData("radio")]
+        [InlineData("submit")]
+        [InlineData("reset")]
+        [InlineData("button")]
+        // Skipping these two because they won't match the constant string,
+        // only because their 'value' attribute also does not get set.
```

Comment:
I take that back; I'm going to amend the tests to simply check for whether the result contains the placeholder attribute rather than comparing against a formatted constant.